homeassistant/components/whirlpool/__init__.py

```diff
@@ -30,7 +38,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         _LOGGER.error("Authentication failed")
         return False
 
-    hass.data[DOMAIN][entry.entry_id] = {AUTH_INSTANCE_KEY: auth}
+    appliances_manager = AppliancesManager(backend_selector, auth)
+    if not await appliances_manager.fetch_appliances():
+        _LOGGER.error("Cannot fetch appliances")
+        return False
+
+    hass.data[DOMAIN][entry.entry_id] = {
+        APPLIANCES_MANAGER_INSTANCE_KEY: appliances_manager,
+        AUTH_INSTANCE_KEY: auth,
+        BACKEND_SELECTOR_INSTANCE_KEY: backend_selector,
+    }
```

Comment:
Instead of this, define a `dataclass`, then you get typing and IDE suggestions when you use it:
`data: WhirlPoolData = hass.data[DOMAIN][entry.entry_id]`

Reply:
done